src/SDKS/erc20/index.ts

```diff
@@ -140,13 +144,33 @@ export namespace CryptoWallet.SDKS.ERC20 {
       });
     }
 
+    public estimateGas(
+      erc20Wallet: any,
+      to: string,
+      amount: number,
+      network: string,
+    ): Promise<number> {
+      const web3: any = new this.Web3(this.networks[network].provider);
+      const contract = new web3.eth.Contract(this.json, erc20Wallet.contract);
+      const sendAmount: string = (amount * (10 ** erc20Wallet.decimals)).toString();
+      return new Promise(async (resolve, reject) => {
+        contract.methods.transfer(to, sendAmount).estimateGas(
+          { from: erc20Wallet.address },
+          (error: Error, gasUsed: number) => {
+            if (error) return reject(error);
+            return resolve(gasUsed);
+          },
+        );
+      });
+    }
+
     /**
      * Create a transaction that transafers ERC20 tokens to a give address
      * @param erc20Wallet
      * @param to
      * @param amount
      */
-    transfer(
+    public transfer(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of public instance method not allowed after declaration of private instance method. Instead, this should come after public instance fields.](https://app.codacy.com/app/atlascity/cryptowallet-js/pullRequest?prid=3649440)